### PR TITLE
✨ Replace flag with pflag

### LINKED ISSUE #106: Switch to pflag package
/kind proposal

***Goals***
Improve and fix the current approach of using flags in the templates by switching to pflag

### PR DESCRIPTION
**Pull request by aniruddha2000**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #106 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -52,26 +53,30 @@ func init() {
...
+	fs.StringVar(&leaderElectionNamespace, "leader-elect-namespace", "", "Namespace that the controller performs leader election in. If unspecified, the controller will discover which namespace it is running in.")
+	fs.IntVar(&hivelocityClusterConcurrency, "hivelocitycluster-concurrency", 1, "Number of HivelocityClusters to process simultaneously")
+	fs.IntVar(&hivelocityMachineConcurrency, "hivelocitymachine-concurrency", 1, "Number of HivelocityMachines to process simultaneously")
+	fs.StringVar(&watchFilterValue, "watch-filter", "", fmt.Sprintf("Label value that the controller watches to reconcile cluster-api objects. Label key is always %s. If unspecified, the controller watches for all cluster-api objects.", clusterv1.WatchLabel))
+	fs.StringVar(&watchNamespace, "namespace", "", "Namespace that the controller watches to reconcile cluster-api objects. If unspecified, the controller watches for cluster-api objects across all namespaces.")
+	fs.StringVar(&logLevel, "log-level", "debug", "Specifies log level. Options are 'debug', 'info' and 'error'")
 
 	flag.Parse()
```
Comment (kranurag7) [inline main.go:81]:
```suggestion
	pflag.Parse()
```
We also have to use `pflag.Parse()` here and remove `flag` altogether.

**Review by kranurag7 — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (aniruddha2000) [reply main.go:81]: Thanks

**Comment by janiskemper** [comment]
do we need to add `	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)` before parsing? I'm not sure...

**Comment by kranurag7** [comment]
> do we need to add ` pflag.CommandLine.AddGoFlagSet(flag.CommandLine)` before parsing? I'm not sure...

Building the executable locally and invoking the same `-h` is working fine. I haven't tested the same on the tilt setup. @aniruddha2000 please test if you have a running tilt setup.

**Comment by aniruddha2000** [comment]
Well, I don't have any access to Hivelocity UI to see if devices are there or not, but I can clearly see that I have providerID in my k9s after creating a cluster by tilt, does that work?

**Comment by janiskemper** [comment]
I think this is more a question of how flags work, not whether the operator works as a whole. But I guess you can proceed

**Review by janiskemper — DISMISSED** [review]

**Review by kranurag7 — DISMISSED** [review]

**Review by janiskemper — APPROVED** [review]

**Pull request merged**